### PR TITLE
Add Notebook for using Elastic Inference with compiled model on SageM…

### PR DESCRIPTION
**Pull request by andrewjguo**
*Issue #, if available:*

*Description of changes:*
* Add notebook for running Neo compiled model with Elastic Inference on SageMaker endpoint, since EI recently released such support. 

Note that this notebook shouldn't be merged until this SageMaker SDK changes are in. https://github.com/aws/sagemaker-python-sdk/pull/2242

I've tested this notebook on my SageMaker Notebook instance successfully.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    "input_model_path = session.upload_data(path='model.tar.gz', bucket=bucket, key_prefix=compilation_job_name)\n",
+    "print('S3 path for input model: {}'.format(input_model_path))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Compile model for EI accelerator using Neo\n",
+    "Now the model is ready to be compiled by Neo. Note that `ml_eia2` needs to be set for `target_isntance_family` field in order for the model to be optimized for EI accelerator. If you want to compile your own model for EI accelerataor, refer to [Neo compilation API](https://sagemaker.readthedocs.io/en/stable/api/inference/model.html#sagemaker.model.Model.compile) to provide the proper `input_shape` and optional `compiler_options` according to your model.\n",
```
Comment (ngluna) [inline sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb:114]: Typo: target_isntance_family -> target_instance_family

File: sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    "                         job_name=compilation_job_name,\n",
+    "                         framework='tensorflow')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Deploy compiled model to SageMaker Endpoint with EI accelerator attached\n",
+    "There is no difference between deploying Neo-compiled model and uncompiled model to EI-attached SageMaker endpoint.\n",
```
Comment (ngluna) [inline sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb:147]: Possible re-wording:
"The same methods are used to deploy a model to a SageMaker Endpoint with EI regardless of whether or not the model is compiled or not compiled by Neo.'

File: sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb [hunk 3]
```diff
@@ -0,0 +1,227 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Using Amazon Elastic Inference with Neo-compiled TensorFlow model on SageMaker\n",
+    "\n",
+    "This notebook demonstrates how to compile a pre-trained TensorFlow model using Amazon SageMaker Neo, and deploy this compiled model on SageMaker with Elastic Inference.\n",
```
Comment (ngluna) [inline sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb:9]: Possible re-wording (feel free to modify):
"This notebook demonstrates how to compile a pre-trained TensorFlow model using Amazon SageMaker Neo and how to deploy this model to a SageMaker Endpoint with Elastic Inference"

File: sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set up the environment\n",
+    "Let's start by creating a SageMaker session and specifying:\n",
+    "\n",
+    "* The S3 bucket that you want to use for model data. This should be within the same region as the Notebook Instance, Neo compilation, and SageMaker hosting.\n",
+    "* The IAM role arn used to give compilation and hosting access to your data. See the [documentation](https://docs.aws.amazon.com/sagemaker/latest/dg/sagemaker-roles.html) for how to create these. Note, if more than one role is required for notebook instances, compilation, and hosting, please replace the sagemaker.get_execution_role() with a the appropriate full IAM role arn string(s)."
```
Comment (ngluna) [inline sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb:36]: Bold face the note, similar to what you did when you wrote "Important: If the following..." below in this notebook.

File: sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
+    "\n",
+    "session = sagemaker.Session()\n",
```
Comment (ngluna) [inline sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb:47]: Add short comments on what each of theses objects do and/or why we are creating them.

Comment (andrewjguo) [reply sagemaker-python-sdk/tensorflow_serving_using_elastic_inference_with_your_own_model/tensorflow_neo_compiled_model_elastic_inference.ipynb:47]: The descriptions for these variables are right above this code block, under "Set up the environment" section

**Review by ngluna — APPROVED** [review]

**Comment by ngluna** [comment]
Dev verified the notebook with latest SM python sdk successfully.

**Pull request merged**